packages/window-size-tool/rules.json

```
@@ -1,52 +1,57 @@
 [
   {
     "width": "< 1200",
-    "terms": ["//CLUSTER/@NUM_COLS > 4", "count(//LIST/*) > 6"],
+    "terms": ["//CLUSTER/@NUM_COLS > 4", "count(//LIST[count(FIELD | CONTAINER) > 6 ]) > 0"
+    ],
     "target": "xlg"
   },
   {
-    "width": ">= 768",
+    "width": "<= 768",
     "terms": [
       "//CLUSTER/@NUM_COLS >= 3",
-      "count(//CLUSTER[@NUM_COLS >= 2]//LIST) > 0",
-      "count(//LIST/*) = 5 or count(//LIST/*) = 6"
+      "count(//CLUSTER[@NUM_COLS > 2]//LIST) > 0",
+      "count(//LIST[count(FIELD | CONTAINER) > 4]) > 0 and count(//LIST[count(FIELD | CONTAINER) < 7]) > 0"
     ],
     "target": "lg"
   },
   {
     "width": "> 768",
     "terms": [
       "//CLUSTER/@NUM_COLS = 2",
```

Firstly

To avoid a conflict with the previous rule you could add a condition here that checks that there are `0` 2 column clusters containing lists.

```suggestion
      "//CLUSTER/@NUM_COLS = 2 and count(//CLUSTER[@NUM_COLS >= 2]//LIST) = 0",
```

but additionally, this rule is just checking for the existance of _any_ 2 column clusters. We want to match where there are _only_ 2 column clusters, so building on the above:

```suggestion
      "//CLUSTER/@NUM_COLS = 2 and count(//CLUSTER[@NUM_COLS >= 2]//LIST) = 0 and count(//CLUSTER[@NUM_COLS > 2]) = 0",
```
we're saying here, if:
- there is at least one 2-column cluster
- there are no 2-column clusters that contain lists
- there are no clusters with more than 2 columns
then change the modal from large to medium

---

Yes that is a better solution for the conflicts, Thanks Graham!